apps/studio/src/components/connection/AddDatabaseForm.vue

```diff
@@ -0,0 +1,77 @@
+<template>
+  <form class="save-connection expand" @submit.prevent="save">
+    <h3 class="dialog-c-title">Add database</h3>
+
+    <div v-if="error" class="alert alert-danger">
+      {{error}}
+    </div>
+
+    <div class="form-group">
+      <input class="form-control" v-model="databaseName" @keydown.enter.prevent.stop="save" type="text" placeholder="Database Name">
+    </div>
+
+    <div class="form-group" v-if="charSets.length > 0">
+      <select v-model="selectedCharset">
+        <option v-for="charset in charSets" v-bind:key="charset" :selected="charset === defaultCharSet" :value="charset">{{charset}}</option>
+      </select>
+    </div>
+
+    <div class="form-group" v-if="collations.length > 0">
+      <select v-model="selectedCollation">
+        <option v-for="collation in collations" v-bind:key="collation" :value="collation">{{collation}}</option>
+      </select>
+    </div>
+
+    <div class="save-actions">
+      <button class="btn btn-flat" @click.prevent="$emit('cancel')">Cancel</button>
+      <button class="btn btn-primary save" type="submit">Add</button>
+    </div>
+  </form>
+</template>
+
+<script>
+  export default {
+    props: ['connection'],
+    data() {
+      return {
+        charSets: [],
+        collations: [],
+        defaultCharSet: null,
+        databaseName: null,
+        selectedCharset: null,
+        selectedCollation: null,
+        error: null
+      }
+    },
+    async mounted(){
+      this.charSets = await this.connection.listCharsets()
+      this.defaultCharSet = await this.connection.getDefaultCharSet()
+      this.selectedCharset = this.defaultCharSet
+      await this.updateCollations()
+    },
+    methods: {
+      async updateCollations() {
+        this.collations = await this.connection.listCollations(this.selectedCharset)
+        this.selectedCollation = this.collations[0]
+      },
+      async save() {
+        const dbNameRegex = /^[a-zA-Z0-9_-]*$/
+        try {
+          if (!this.databaseName.match(dbNameRegex)) {
+            throw new Error('Database name invalid, must be alphanumeric / have only _ or - special characters')
```

Comment:
I'm not sure this is always true? Can special characters in database names be escaped like table names in some databases?

Is there a reason we can't just rely on the database creation function to provide us with an error here?

Reply:
I was basing it largely on the postgres rules since that's where I was having the most issue. Totally fine with knocking it back to the database creation function to be the final word, was thinking from a user perspective give them something prior, but I get the issue. I can take this out no problem.